### DISCUSSION TITLE
How to detect first moved piece?

### DISCUSSION THREAD

**Discussion by sebastiaandegeus** [flbulgarelli/headbreaker#58] [Q&A] [answered]
Hi,

I would like to start a timer once the first piece is moved. How could I do that? 

Thank you

**Comment by flbulgarelli** [comment, marked as answer]
Hi @sebastiaandegeus 

I suggest registering a `TranslationListener` to each piece of a puzzle or canvas (https://flbulgarelli.github.io/headbreaker/api/Piece.html) using `onTranslate`. Then you can use a global flag to detect if that is the first movement and your timer should be started.

Hope that helps!

**Comment by sebastiaandegeus** [comment]
Thank you it works. :)
